### ISSUE TITLE
"fork.settings" does not exist when running ForkCMSInstallerBundle

### ISSUE THREAD

**Issue by jeroendesloovere** [forkcms/forkcms#1407]
Hi, I'm trying to implement the MailMotor into Fork CMS.

### Steps to reproduce
* Create custom module
* Add /Resources/config/services.yml with "@fork.settings" as an attribute
* Try to install Fork CMS with this module enabled OR try to deploy using Capistrano

### Problem
But when deploying using capistrano, I get the following error log:
```bash
*** [err :: xxx.be] Generating optimized autoload files
*** [err :: xxx.be] > php app/console assetic:dump . && cd tools && bash ./remove_cache
 ** [out :: xxx.be] PHP Fatal error:  Uncaught exception 'Symfony\Component\DependencyInjection\Exception\InvalidArgumentException' with message 'The service definition "fork.settings" does not exist.' in /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:815
 ** [out :: xxx.be] Stack trace:
 ** [out :: xxx.be] #0 /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php(459): Symfony\Component\DependencyInjection\ContainerBuilder->getDefinition('fork.settings')
 ** [out :: xxx.be] #1 /sites/web/apps/staging/releases/20160224150219/src/Backend/Modules/MailMotor/DependencyInjection/Compiler/CustomCompilerPass.php(18): Symfony\Component\DependencyInjection\ContainerBuilder->get('fork.settings')
 ** [out :: xxx.be] #2 /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php(107): Backend\Modules\MailMotor\DependencyInjection\Compiler\CustomComp in /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php on line 815
*** [err :: xxx.be] Script php app/console assetic:dump . && cd tools && bash ./remove_cache handling the post-install-cmd event returned with an error
*** [err :: xxx.be] 
*** [err :: xxx.be] 
*** [err :: xxx.be] [RuntimeException]
*** [err :: xxx.be] Error Output: PHP Fatal error:  Uncaught exception 'Symfony\Component\DependencyInjection\Exception\InvalidArgumentException' with message 'The service definition "fork.settings" does not exist.' in /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php:815
*** [err :: xxx.be] Stack trace:
*** [err :: xxx.be] #0 /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php(459): Symfony\Component\DependencyInjection\ContainerBuilder->getDefinition('fork.settings')
*** [err :: xxx.be] #1 /sites/web/apps/staging/releases/20160224150219/src/Backend/Modules/MailMotor/DependencyInjection/Compiler/CustomCompilerPass.php(18): Symfony\Component\DependencyInjection\ContainerBuilder->get('fork.settings')
*** [err :: xxx.be] #2 /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Compiler/Compiler.php(107): Backend\Modules\MailMotor\DependencyInjection\Compiler\CustomComp in /sites/web/apps/staging/releases/20160224150219/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ContainerBuilder.php on line 815
*** [err :: xxx.be] 
*** [err :: xxx.be] 
*** [err :: xxx.be] install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--] [<packages>]...
*** [err :: xxx.be] 
*** [deploy:update_code] rolling back
```

In the past I sometimes had problems with "fork cms services" (which are not bundles) that they don't appear in ```app/console container:debug```. Related https://github.com/forkcms/forkcms/issues/1330

### Error triggered by

In composer.json and composer.lock, the following triggers the error and the rollback.
```json
php app/console assetic:dump
```

### Temporary solution

We must remove```php app/console assetic:dump``` in ```composer.json```
```json
    "scripts": {
        "post-install-cmd": "php app/console assetic:dump . && cd tools && bash ./remove_cache",
        "post-update-cmd": "php app/console assetic:dump . && cd tools && bash ./remove_cache"
    },
```
to
```json
    "scripts": {
        "post-install-cmd": [
            "cd tools && bash ./remove_cache"
        ],
        "post-update-cmd": [
            "cd tools && bash ./remove_cache"
        ]
    },
```
> Don't forget to do ```composer update``` so the composer.json is updated to.

Also change the following rule in ```config/defaults/config.rb```
```bash
run "cd #{release_path} && #{composer_bin} install -o --no-dev"
```
to
```
run "cd #{release_path} && #{composer_bin} install -o --dev"
```
> Then the errors will still show, but there will be no rollback.

Don't forget to merge the fix with your github staging and production repo branch.

**Comment by carakas** [comment]
Are your database settings correct ?
If they are not correct the container won't be build, and the services won't be available.
This is so only services of installed modules are loaded

**Comment by jeroendesloovere** [comment]
My local website is working perfect!
=> No service definition errors!

I only get this message when using "cap staging deploy".

**Possible problem**
When using capistrano our fork modules "not bundles" are doing something wrong and aren't being defined.

**Comment by jeroendesloovere** [comment]
I've added the problem and the (temporary) solution...

**Comment by jeroendesloovere** [comment]
I'm integrating the new MailMotor in 3.10 and when trying to install Fork CMS, I get the same error:
```bash
The service definition "fork.settings" does not exist.
```


<img width="746" alt="schermafbeelding 2016-02-28 om 10 17 40" src="https://cloud.githubusercontent.com/assets/588616/13378367/727a3032-de04-11e5-968e-b0e4f72f666b.png">

**Comment by jeroendesloovere** [comment]
I've added an if-statement in [CustomCompilerPass ](https://github.com/forkcms/forkcms/pull/1413/files#diff-119759ce9d3ef9cf33fd708c0c9c5cf9R16) to check if "fork.settings" exist (because "fork.settings" does not exist, when installing the module).

**Comment by jeroendesloovere** [comment]
@WouterSioen Any news about fixing this in Fork CMS?
"fork.settings" service should be available in the fork installer (because modules can rely on this).
(the service will be empty, since Fork is not yet installed at this moment)

**Comment by carakas** [comment]
this will be fixed when #1632 lands
it adds 
```
services:
    # dummy service during the installation
    fork.settings:
        class: stdClass
```
in `config_test.yml`

**Comment by WouterSioen** [comment]
fixed in #1632
